compiler/rustc_hir_typeck/src/method/mod.rs

```diff
@@ -91,7 +91,7 @@ pub enum CandidateSource {
 impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
     /// Determines whether the type `self_ty` supports a visible method named `method_name` or not.
     #[instrument(level = "debug", skip(self))]
-    pub fn method_exists(
+    pub fn method_exists_for_diagnostic(
```

Comment:
I find the `for_diagnostic` suffix naming convention confusing. I think it means "method exists (as considered for diagnostic purposes)", rather than "method exists for this diagnostic"? It feels like there should be a contrasting `method_exists_for_non_diagnostic` method. I'm just not sure how to read it.

Reply:
Yes, it means "and you should only call this method if you're doing diagnostics".

Reply:
Kinda don't want to change it though now lol

Reply:
Ok, I can live with it.